### ISSUE TITLE
Allow "--offset-days" before removing via sync/delete/prune

### ISSUE THREAD

**Issue by wellloaded** [rclone/rclone#7303]
I had a quick look at the documentation but I don't believe this has been implemented yet. Some cloud providers allow to recover deleted files which I find extremely helpful but I also admit it might have a huge impact in terms of additional storage/traffic needed to handle the feature. What I currently do is to perform a daily copy local->remote and once a month a sync local->remote. However this is static and if I delete something by accident on the 29th I have 1 or 2 days available to restore before the sync wipes from the remote.

What I thought about as a feature request is more a poor man solution achieving hopefully the same a the enterprise solution described above. The scenario is simple: keep files on the target up to N day (customizable) after they were removed from the source. Say 180 by default just to give a number.

This allows for e.g. a sync to happen daily between local and remote with the locally removed file remaining available for restore on the target for 180 days. Somehow the deletion needs to be recorded in a log or something on both sides I would have though, also there should be a command to display what is currently restorable content that has been deleted within the N days windows.

Once this is done, say that at some point in the future the source disappears due to a HW failure; it would be nice to have this offset log consulted when sync/copy back from remote to local e.g. ignore (do not restore) files that were previously deleted even if still present on remote but keep the restore log in case I need to restore something later on.

I'm not going into the file versioning as it's a completely different issue although could be nice to have. For the sake of this feature request I would just handle file/folder deletion.

**Comment by animosity22** [comment]
I can't quite figure out what you want to happen, but there is already is max-age and min-age which allows for this.

https://corespecialty.atlassian.net/browse/AM-4065

**Comment by wellloaded** [comment]
> I can't quite figure out what you want to happen, but there is already is max-age and min-age which allows for this.
I would like to restore files on local after they have been deleted using the remote location as a source for the restore.

> https://rclone.org/filtering/#max-age-don-t-transfer-any-file-older-than-this
Does this work with deleted files?

**Comment by animosity22** [comment]
I don't know as you neglected to use the template which collects a bunch of information as it heavily depends on the backend on how to answer that question since one size doesn't fit all.

On most remotes, a deleted file is 'gone' so nothing would work against it.

**Comment by ncw** [comment]
You can build something like this using --backup-dir and/or --suffix 

Then supply the current date as an argument.

You can then write a little script to expire the files using the suffix.

**Comment by wellloaded** [comment]
> You can build something like this using --backup-dir and/or --suffix
> 
> Then supply the current date as an argument.
> 
> You can then write a little script to expire the files using the suffix.

Thanks this might be a way and I will have a look at it. Regardless is there any interest at all to have an internal function covering this? I mean rclone built-in "content deletion recovery".

Thanks

**Comment by kapitainsky** [comment]
What you are asking for can be easily achieved by using remote with versioning (many S3 implementations (AWS, iDrive, Wasabi), B2, Azure, etc.). 

You can always "go back in time" and use remote state from given date using [--s3-version-at](https://rclone.org/s3/#s3-version-at) flag, e.g.:

`rclone ls s3remote: --s3-version-at 2023-02-25 18:00:00` will list remote content as it was at `2023-02-25 18:00:00`

You can also use [s3-versions](https://rclone.org/s3/#s3-versions) flag to list all stored versions and delete versions older than given date.

**Comment by animosity22** [comment]
This seems answered a few ways so closing it out. Let me know if we missed something and happy to open it back up.
